sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ReshuffleTest.java

```diff
@@ -70,9 +73,16 @@
         KV.of("k1", 3),
         KV.of("k2", 4));
 
-  private static final List<KV<String, Iterable<Integer>>> GROUPED_TESTABLE_KVS = ImmutableList.of(
-        KV.of("k1", (Iterable<Integer>) ImmutableList.of(3)),
-        KV.of("k2", (Iterable<Integer>) ImmutableList.of(4)));
+  private static class AssertThatHasExpectedContents
```

Comment:
<!--new_thread; commit:9f415d4d04ecbb6a985de9e9a27f27e427694877; resolved:0-->
I think you can just switch this to be a list of matchers so you can say

PAssert.that(collection).containsInAnyOrder(isKv(is("k1"), containsInAnyOrder(3)), ...)

and so on.